src/KubeOps/Operator/Controller/ManagedResourceController{TEntity}.cs

```diff
@@ -309,6 +310,8 @@ protected async Task HandleResourceEvent(QueuedEvent? data)
 
         protected async Task HandleResourceFinalization(QueuedEvent? data)
         {
+            using var scope = _services.CreateScope();
```

Comment:
I think some parts were maybe; there were a handful of `CreateScope` calls from other places that I removed (don't remember exactly where off-hand) but in the interest of not referencing `IServiceProvider` in every third class, I consolidated that as much as possible. `ManagedResourceController` creates a scope whenever handling a controller event or finalizing a resource, and webhooks automatically get a scope because they come in as web requests.